src/Agent.Worker/RetryHelper.cs

```diff
@@ -48,6 +62,55 @@ public async Task<T> Retry<T>(Func<Task<T>> action, Func<int, int> timeDelayInte
 
         }
 
+        /// <summary>
+        /// Runs action with maxRetries number of retries
+        /// </summary>
+        /// <param name="action">Action to execute with retries</param>
+        /// <param name="timeDelayInterval">Function to calculate delay between retries depending on retry number. Should take retry number as argument and consider that it starts from 0.</param>
+        /// <returns></returns>
+        public async Task RetryStep(Func<Task> action, Func<int, int> timeDelayInterval)
+        {
+            int retryCounter = 0;
+            do
+            {
+                using (new SimpleTimer($"RetryHelper Method:{action.Method} ", Debug))
+                {
+                    var delayInterval = timeDelayInterval(retryCounter);
+                    try
+                    {
+                        if (retryCounter > 0)
+                        {
+                            ExecutionContext.ReInitializeForceCompleted();
+                        }
+
+                        Debug($"Invoking Method: {action.Method}. Attempt count: {retryCounter}");
+                        await action();
+
+                        if (ExecutionContext.Result != TaskResult.Failed || ExhaustedRetryCount(retryCounter))
+                        {
+                            return;
+                        }
+                        else
+                        {
+                            string exceptionMessage = $"Task result ${ExecutionContext.Result}";
+                            ExecutionContext.Result = null;
+                            Warning($"RetryHelper encountered task failure, will retry (attempt #: {retryCounter + 1} out of {this.MaxRetries}) after ${delayInterval} ms");
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        if (!ShouldRetryStepOnException(ex) || ExhaustedRetryCount(retryCounter))
+                        {
+                            throw;
+                        }
+                        Warning($"RetryHelper encountered exception, will retry (attempt #: {retryCounter} {ex.Message}) afer ${delayInterval} ms");
```

Comment:
Considering that the initial value of the `retryCounter` is 0 could it make sense to apply + 1 to it in the message? To avoid confusing customers with the retry counter?